packages/iTwinUI-react/src/core/Table/Table.test.tsx

```diff
@@ -2883,6 +2883,66 @@ it('should render dropdown menu with custom styling and override default styling
   expect(dropdownMenu).toHaveStyle('background-color: red');
 });
 
+it('should have all props other than className and style be also passed from ActionColumn to DropdownMenu', async () => {
+  const columns: Column<TestDataType>[] = [
+    {
+      Header: 'Header name',
+      columns: [
+        {
+          id: 'name',
+          Header: 'Name',
+          accessor: 'name',
+        },
+        {
+          id: 'description',
+          Header: 'Description',
+          accessor: 'description',
+        },
+        {
+          id: 'view',
+          Header: 'View',
+          Cell: () => <>View</>,
+        },
+        ActionColumn({
+          columnManager: {
+            dropdownMenuProps: {
+              className: 'testing-classname',
+              style: {
+                maxHeight: '500px',
+                backgroundColor: 'purple',
+              },
+              role: 'listbox', // Sample property other than className and style
```

Comment:
You could just include this in the other test, but upto you

Reply:
Oh that's a good point. When passing props to a react component in JSX, the data attributes are automatically allowed, but since we are accepting these props as a plain object, TS can't make the connection.

I think the problem might be that `TippyProps` (which `PopoverProps` is based on) doesn't extend `React.ComponentProps`. I wonder if we should try to find a "proper" fix or just allow any arbitrary props using `[k: string]: unknown`. @veekeys What do you think?

Reply:
Hmm... We have few components, which do not extend all native elements props and we never really agreed, which path we should follow
- Just extend native elements props on root element
- Allow to pass any prop with `[k: string]: unknown`
- Provide a separate prop for native props, not to flood with all the options so its hard to find our custom props in a huge list

I think we have all 3 options used on different parts in the library.
`[k: string]: unknown` option would seem the easiest if we want to cover only `data` props right now.

Reply:
> Oh that's a good point. When passing props to a react component in JSX, the data attributes are automatically allowed, but since we are accepting these props as a plain object, TS can't make the connection.
> 
> I think the problem might be that `TippyProps` (which `PopoverProps` is based on) doesn't extend `React.ComponentProps`. I wonder if we should try to find a "proper" fix or just allow any arbitrary props using `[k: string]: unknown`. @veekeys What do you think?

I think I misunderstood this React concept, because I tried passing a testing data value to `columnManager`, `dropdownMenu`, or even directly `ActionColumn`, but I couldn't see the data value being added to the HTML tag. To just test this concept of automatically adding data props, I tried it in a simple [React app](https://codesandbox.io/s/react-playground-forked-q09bbh?file=/index.js) on CodeSandbox. I unfortunately couldn't do it there too. @mayank99 @veekeys Can you please help?

![image](https://user-images.githubusercontent.com/45748283/183896467-4b3b3c15-a5cc-448f-9bba-b6f038365b21.png)

Reply:
In that example you are not spreading props from `MyComp` in the rendered `<div>`. It should show up correctly if you do it like this:
```js
const MyComp = (props) => {
 return <div {...props}>MyComp123</div>;
};
```

Reply:
**[DONE]** Thanks. After taking help for Vykintas, we found out that passing the data prop and adding it to the HTML element is not possible due to the following:

* After going up the hierarchy, we noticed that there is a Tippy element (external library) that may not include all remaining props `{...rest}` in the attributes list.
* We also might need to change our `Menu` component, which is not intended in this PR

So for now, the new test just tests for `role: 'listbox'` and not for any data properties.